### ISSUE TITLE
feat: add class_weight and L2 regularization to LogisticRegression

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#427]
## Context

bashrs SSC classifier pipeline (CLF-RUN) uses a linear probe on CodeBERT [CLS] embeddings for binary shell safety classification. The corpus is heavily imbalanced (~92% safe, ~8% unsafe). Without class weighting, the probe converges to predicting "safe" for everything at scale (MCC drops from 0.59 at 500 entries to 0.20 at 1675 entries).

## Problem

`aprender::classification::LogisticRegression::fit()` has a hand-rolled gradient descent loop that:
1. Does NOT support class weighting (needed for imbalanced data)
2. Does NOT support L2 regularization (weight decay)

Meanwhile, aprender already has the primitives:
- `BCEWithLogitsLoss` with `pos_weight` parameter
- `SGD` optimizer with `weight_decay` support

But `LogisticRegression.fit()` doesn't compose these.

## Proposed API

```rust
let mut model = LogisticRegression::new()
    .with_learning_rate(0.001)
    .with_max_iter(200)
    .with_class_weight(ClassWeight::Balanced)  // NEW: sqrt-inverse auto-weighting
    .with_l2_penalty(1e-4);                    // NEW: L2 regularization

model.fit(&x, &y)?;
```

### ClassWeight enum

```rust
pub enum ClassWeight {
    /// No class weighting (default, backward compatible)
    Uniform,
    /// Automatic sqrt-inverse weighting: w_k = sqrt(n_total / (n_classes * n_k))
    Balanced,
    /// Manual per-class weights: [w_0, w_1]
    Manual(Vec<f32>),
}
```

### Implementation

In `fit()`, multiply each sample's gradient by its class weight:
```rust
let class_w = match &self.class_weight {
    ClassWeight::Uniform => 1.0,
    ClassWeight::Balanced => balanced_weights[label],
    ClassWeight::Manual(w) => w[label],
};
let error = class_w * (prob - target);
```

For L2 regularization, add weight decay to coefficient update:
```rust
coef[j] -= lr * (grad[j] + self.l2_penalty * coef[j]);
```

## Downstream Consumer

bashrs `rash/src/corpus/classifier.rs:train_linear_probe()` currently hand-rolls class-weighted SGD with L2. This ticket would let us replace ~100 lines with:

```rust
use aprender::classification::{LogisticRegression, ClassWeight};

let mut model = LogisticRegression::new()
    .with_learning_rate(lr)
    .with_max_iter(epochs)
    .with_class_weight(ClassWeight::Balanced)
    .with_l2_penalty(1e-4);

model.fit(&embedding_matrix, &labels)?;
model.save_safetensors("probe.safetensors")?;
```

## Acceptance Criteria

- [ ] `ClassWeight` enum with `Uniform`, `Balanced`, `Manual` variants
- [ ] `with_class_weight()` builder method
- [ ] `with_l2_penalty()` builder method  
- [ ] Balanced weights computed as `sqrt(n_total / (n_classes * n_k))`
- [ ] Gradient scaled by class weight in `fit()` loop
- [ ] L2 penalty applied as weight decay in coefficient update
- [ ] Backward compatible: default behavior unchanged (Uniform + no penalty)
- [ ] Tests: class-weighted training on imbalanced synthetic data
- [ ] Tests: L2 regularization reduces coefficient magnitudes
- [ ] Existing tests pass unmodified

## References

- SSC v11 Section 4.3: Classifier Infrastructure
- `provable-contracts/contracts/classifier-pipeline-v1.yaml`
- Alain & Bengio (2016) linear probe methodology
- scikit-learn `class_weight='balanced'` reference implementation
